src/components/OptionsBar/Options/FilterModal/Filters/RelativeFilters.jsx

```diff
@@ -4,43 +4,43 @@ import { useRef } from 'react';
 const relativeFilters = (filterIds, tabIndex) => {
   const filters = useSelector((state) => state.filters);
 
-  const getFilterMatrix = () => {
-    const filterMatrix = new Array(filterIds.length);
-
-    filterIds.forEach((id, index) => {
-      filterMatrix[index] = filters[id];
-    });
-
-    return filterMatrix;
-  };
+  const getFilterMatrix = () => (filters ? filterIds.map((id) => filters?.[id]) : []);
 
   const queuedFilters = useRef(getFilterMatrix());
 
   const getFilters = () => queuedFilters.current;
 
   const getRelativeFilters = () => getFilters()[tabIndex];
 
-  const pushRelativeFilters = (filter) => getRelativeFilters().push(filter);
+  const setRelativeFilters = (value) => {
+    queuedFilters.current[tabIndex] = value;
+
+    return getRelativeFilters();
+  };
 
-  const spliceRelativeFiltersAt = (index) => getRelativeFilters().splice(index, 1);
+  const pushRelativeFilters = (filter) => {
+    setRelativeFilters([ ...getRelativeFilters(), filter ]);
```

Comment:
It left me to wonder why it was returning not the value you came up with, but then I understood why. I tink it would be more clear if you would separate the statements, and return what you built 

```suggestion
    const extendedRelativeFilters = [ ...getRelativeFilters(), filter ];
    
    setRelativeFilters(extendedRelativeFilters);
    
    return extendedRelativeFilters;
```